crypto/evp/pmeth_fn.c

```diff
@@ -18,7 +18,12 @@
         if (ctx->pmeth->flags & EVP_PKEY_FLAG_AUTOARGLEN) \
                 { \
                 size_t pksize = (size_t)EVP_PKEY_size(ctx->pkey); \
-                if (!arg) \
+                if (pksize == 0) \
```

Comment:
the curly braces are misplaced.  since you're changing the macro, how do you feel about not making the situation worse and fixing this, too?